e2e/security/security.go

```diff
@@ -0,0 +1,292 @@
+		image         string
+		action        string
+		expectID      string
+		argv          []string
+		opts          e2e.ExecOpts
+		exit          int
+		expectSuccess bool
+	}{
+		// taget UID/GID
+		{
+			name:          "Set_uid",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"id", "-u"},
+			opts:          e2e.ExecOpts{Security: []string{"uid:99"}},
+			expectID:      "99",
+			exit:          0,
+			expectSuccess: true,
+		},
+		{
+			name:          "Set_gid",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"id", "-g"},
+			opts:          e2e.ExecOpts{Security: []string{"gid:99"}},
+			expectID:      "99",
+			exit:          0,
+			expectSuccess: true,
+		},
+		// seccomp from json file
+		{
+			name:          "SecComp_BlackList",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"mkdir", "/tmp/foo"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "SecComp_true",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"true"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          0,
+			expectSuccess: true,
+		},
+		// capabilities
+		{
+			name:          "capabilities_keep",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:          e2e.ExecOpts{KeepPrivs: true},
+			exit:          0,
+			expectSuccess: true,
+		},
+		{
+			name:          "capabilities_drop",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:          e2e.ExecOpts{DropCaps: "CAP_NET_RAW"},
+			exit:          1,
+			expectSuccess: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("priv "+tt.name, test.WithPrivilege(func(t *testing.T) {
+			stdout, stderr, exitCode, err := e2e.ImageExec(t, testenv.CmdPath, tt.action, tt.opts, tt.image, tt.argv)
+
+			lines := bytes.Fields([]byte(stdout))
+
+			if tt.expectSuccess && (exitCode != 0) {
+				t.Log(stderr, err, exitCode)
+				t.Fatalf("unexpected failure running '%v': %v", strings.Join(tt.argv, " "), err)
+			} else if !tt.expectSuccess && (exitCode != 1) {
+				t.Log(stderr, err, exitCode)
+				t.Fatalf("unexpected success running '%v'", strings.Join(tt.argv, " "))
+			}
+			//if len(lines) != 1 || string(lines[0]) != "99" {
+			if len(lines) == 1 && string(lines[0]) != tt.expectID {
+				t.Fatal("test failed? expecting: 99, got: ", string(lines[0]))
+			}
+
+		}))
+	}
+}
+
+// testSecurityConfOwnership tests checks on config files ownerships
+func testSecurityConfOwnership(t *testing.T) {
+	configFile := buildcfg.SYSCONFDIR + "/singularity/singularity.conf"
+	// Change file ownership (do not try this at home)
+	err := os.Chown(configFile, 1001, 0)
```

Comment:
probably better to use "nobody" instead of 1001.

For example, in my specific case my UID is in fact 1001.